Fixed TestDisconnection

russjones opened the pull request:
Stabalized `TestDisconnection` and `TestSessionStartContainsAccessRequest`.

**TestDisconnection**

Integration tests were creating the users SSH and x509 certificates at the time of server creation, specifically at the time of the `TeleInstance` creation.

If any component within the `TeleInstance` was slow to start, for example if the proxy watcher had to be reset as can be seen below, the users certificate would often be expired by the time the SSH request was issued which would lead to the SSH server rejecting the connection and the test failing with an unexpected error.

For most tests this is not an issue, because test certificates are valid for 24 hours. However `TestDisconnection` scopes the TTL of certificates down (often times to about 2 seconds) to test different certificate lifetime disconnection scenarios which would lead to this test being flaky.

Updated integration test logic to instead "issue" the users certificate at the time of client creation instead of server creation.

**TestSessionStartContainsAccessRequest**

Fixed issue with `TestSessionStartContainsAccessRequest` where the login was not being injected into the user role.

russjones committed:
Fixed TestDisconnection

Integration tests were creating the users SSH and x509 certificates at
the time of server creation, specifically at the time of the
"TeleInstance" creation.

If any component within the "TeleInstance" was slow to start, for example
if the proxy watcher had to be reset as can be seen below, the users
certificate would often be expired by the time the SSH request was issued
which would lead to the SSH server rejecting the connection and the test
failing with an unexpected error.

For most tests this is not an issue, because test certificates are valid
for 24 hours. However "TestDisconnection" scopes the TTL of certificates
down (often times to about 2 seconds) to test different certificate
lifetime disconnection scenarios which would lead to this test being flaky.

Updated integration test logic to instead "issue" the users certificate at
the time of client creation instead of server creation.

russjones committed:
Fixed TestSessionStartContainsAccessRequest.

Fixed issue with TestSessionStartContainsAccessRequest where the login
was not being injected into the user role.

zmb3 commented on integration/helpers.go:
```diff
@@ -1228,30 +1214,30 @@ func (i *TeleInstance) NewUnauthenticatedClient(cfg ClientConfig) (tc *client.Te
 // NewClient returns a fully configured and pre-authenticated client
-// (pre-authenticated with server CAs and signed session key)
+// (pre-authenticated with server CAs and signed session key).
 func (i *TeleInstance) NewClient(t *testing.T, cfg ClientConfig) (*client.TeleportClient, error) {
```
I find it odd for a func to accept a `testing.T` **and** return an error.

If there's an error, shouldn't it just fail the test?

russjones replied:
Just saw that, updated to no longer accept `testing.T` and instead return an error and have the caller make the decision.

zmb3 reviewed (approved):
Looks fine to me.

Small comment not related to the change here but worth considering either way.

russjones committed:
Cleaned up NewClient in integration tests.

Updated "NewClient" in integration tests to not take testing.T and
return an error.
